### ISSUE TITLE
Config property is undefined when using promise wrapper.

### ISSUE THREAD

**Issue by Ebuall** [sidorares/node-mysql2#804]
Config property is undefined when using promise wrapper.
I see there is an underlying connection attached at "connection" property, but it breaks compatibility with typings and with mysqljs/mysql

**Comment by sidorares** [comment]
can you show some code to explain better problem?
Promise wrapper is not always 1:1 compatible, but when make sense it should be

**Comment by Ebuall** [comment]
```typescript
promiseConnection.connection.config // { host, user, database... }
promiseConnection.config // undefined, even though is defined in the interface

// but the first in typescript is looks like
(promiseConnection as any).connection.config // discouraged and unsafe
```
Of course you might think of it as a typing issue

**Comment by sidorares** [comment]
where do you get typings from? I don't mind adding `.config` to promise wrapper as well but interesting to understand why your setup expected to see this property

**Comment by wellwelwel** [comment]
This has been fixed in #1610 🙋🏻‍♂️

https://github.com/sidorares/node-mysql2/blob/4ac6a8f0f908cfa7af1f2dddd17e4054af73e52f/promise.d.ts#L43
